expirationd.lua

```diff
@@ -144,12 +155,12 @@ local function worker_loop(task)
     fiber.name(string.format("worker of %q", task.name), { truncate = true })
 
     while true do
-        if task.worker_canceled then
-            fiber.self():cancel()
-        end
         if (box.cfg.replication_source == nil and box.cfg.replication == nil) or task.force then
             task.on_full_scan_start(task)
             local state, err = pcall(task.do_worker_iteration, task)
+            -- Following functions are on_full_scan*,
+            -- but we probably did not complete the full scan,
+            -- so we should logically check cancellation here.
```

Comment:
It's a bit strange phrase.

Could we perform "non-logical" check? Also I guess it should be "check for cancellation".

Reply:
Did this:
```diff
             local state, err = pcall(task.do_worker_iteration, task)
             -- Following functions are on_full_scan*,
             -- but we probably did not complete the full scan,
-            -- so we should logically check cancellation here.
+            -- so we should check for cancellation here.
             if task.worker_canceled then
                 fiber.self():cancel()
             end
```